### PR TITLE
Removing BindPFlag method from viper

### PR DESCRIPTION
**Pull request by hush-hush**
### What does this PR do?

In order to simplify the configuration migration we are working on removing the use of unwanted methods. BindPFlag goes against the standardized way to overwrite configuration through env vars.

This also removes 'cmd.check.fullsketches' settings since it wasn't used anywhere.

### Describe how to test/QA your changes

Starting dogstatsd with a unix socket and checking it's correctly used.

### CODE REVIEW

File: pkg/cli/subcommands/check/command.go [hunk 1]
```diff
@@ -236,14 +235,11 @@ func MakeCommand(globalParamsGetter func() GlobalParams) *cobra.Command {
 	cmd.Flags().BoolVarP(&cliParams.formatTable, "table", "", false, "format aggregator and check runner output as an ascii table")
 	cmd.Flags().StringVarP(&cliParams.breakPoint, "breakpoint", "b", "", "set a breakpoint at a particular line number (Python checks only)")
 	cmd.Flags().BoolVarP(&cliParams.profileMemory, "profile-memory", "m", false, "run the memory profiler (Python checks only)")
-	cmd.Flags().BoolVar(&cliParams.fullSketches, "full-sketches", false, "output sketches with bins information")
```
Comment (vickenty) [inline pkg/cli/subcommands/check/command.go:239]: We need to fix the unusedness of this option, but even if we didn't need to support it anymore, the check command still needs to accept it for backwards compatibility. Would it be possible to keep this line?

Comment (hush-hush) [reply pkg/cli/subcommands/check/command.go:239]: good point about backward compatibility ! I'll add it back

**Review by vickenty — APPROVED** [review]

File: pkg/cli/subcommands/check/command.go [hunk 2]
```diff
@@ -104,7 +104,6 @@ type cliParams struct {
 	formatJSON                bool
 	formatTable               bool
 	breakPoint                string
-	fullSketches              bool
```
Comment (dustmop) [inline pkg/cli/subcommands/check/command.go:107]: Need to remove line 239 as well `cmd.Flags().BoolVar(&cliParams.fullSketches, ...`

**Review by dustmop — APPROVED** [review]
LGTM, one fix needed

**Pull request merged**